### PR TITLE
[SYCL] Added a new category for device binary properties

### PR DESCRIPTION
**Pull request by DenisBakhvalov**
This PR introduces a new category for general properties of device binary images. And it adds an API to retrieve properties from that common category.

### CODE REVIEW

File: sycl/source/detail/pi.cpp
Comment (kbobrovs) [inline sycl/source/detail/pi.cpp:606]: Is it possible to write a unit test (similar to https://github.com/intel/llvm/blob/sycl/sycl/unittests/kernel-and-program/KernelRelease.cpp) which checks if the new property is recognized?

**Review by smaslov-intel — DISMISSED** [review]
*pi* changes look good to me

**Review by bader — DISMISSED** [review]
llvm/* part looks good to me.

File: sycl/source/detail/pi.cpp
Comment (DenisBakhvalov) [reply sycl/source/detail/pi.cpp:606]: Seems that it is not that easy. In order to test this code, we need to construct an in-memory object of  `pi_device_binary_struct`, which then would be parsed by `DeviceBinaryImage` class. The `clang-offload-wrapper` tool seems to do similar things, but there is no code we can directly reuse.

Comment (bader) [reply sycl/source/detail/pi.cpp:606]: @kbobrovs, the patch is approved by code owners.
Could you mark this conversation as resolved if your comment is addressed, please?

**Review by kbobrovs — DISMISSED** [review]

File: sycl/include/CL/sycl/detail/pi.hpp
Comment (romanovvlad) [inline sycl/include/CL/sycl/detail/pi.hpp:336]: It looks like a backward incompatible change to me as this class crosses library boundaries.
Can we instead of adding a new field to the class search for `__SYCL_PI_PROPERTY_SET_SYCL_IS_ESIMD_IMAGE` in the `Bin` in the `getSyclIsEsimdImage` method(like it's done in the `PropertyRange::init` method)?

tagging @kbobrovs

**Review by romanovvlad — CHANGES_REQUESTED** [review]

File: sycl/include/CL/sycl/detail/pi.hpp
Comment (DenisBakhvalov) [reply sycl/include/CL/sycl/detail/pi.hpp]: In fact, I don't need `SyclIsEsimdImage` to be a member of the `DeviceBinaryImage` class. Fixed.

**Review by romanovvlad — APPROVED** [review]

**Comment by DenisBakhvalov** [comment]
I think I have all the approvals here. @bader or @kbobrovs, please merge.

**Review by kbobrovs — APPROVED** [review]

File: llvm/include/llvm/Support/PropertySetIO.h
Comment (kbobrovs) [inline llvm/include/llvm/Support/PropertySetIO.h:190]: @DenisBakhvalov, as we agreed, can you please change the category name for this to something like SYCL/misc or SYCL/binary image?

**Review by kbobrovs — CHANGES_REQUESTED** [review]

**Comment by kbobrovs** [comment]
> I think I have all the approvals here. @bader or @kbobrovs, please merge.

@DenisBakhvalov, please see comment above. If believe the category name change is still pending, right?

**Comment by DenisBakhvalov** [comment]
> > I think I have all the approvals here. @bader or @kbobrovs, please merge.
> 
> @DenisBakhvalov, please see comment above. If believe the category name change is still pending, right?

@kbobrovs Yes, I'm working on it.

**Review by kbobrovs — APPROVED** [review]
Thanks.

**Review by romanovvlad — APPROVED** [review]

**Comment by dendibakh** [comment]
@bader @kbobrovs , please merge if no objections.

**Comment by bader** [comment]
image

The approval from @smaslov-intel is required. There were changes in the code he owns since his latest review.

**Review by smaslov-intel — APPROVED** [review]
LGTM

**Pull request merged**
